crates/solver/src/settlement_simulation.rs

```diff
@@ -119,6 +124,75 @@ pub async fn simulate_and_error_with_tenderly_link(
         .collect()
 }
 
+#[derive(Debug, Clone, Deserialize)]
+struct TenderlyResponse {
+    transaction: TenderlyTransaction,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+struct TenderlyTransaction {
+    gas_used: u64,
+}
+
+pub async fn simulate_before_after_access_list(
+    web3: &Web3,
+    tenderly: &TenderlyApi,
+    network_id: String,
+    transaction_hash: H256,
+) -> Result<f64> {
+    let transaction = web3
+        .eth()
+        .transaction(transaction_hash.into())
+        .await?
+        .ok_or_else(|| anyhow!("no transaction found"))?;
```

Comment:
ubernit: Some `anyhow` niceties are slightly more ergonomic.
```suggestion
        .context("no transaction found")?;
```

Not blocking PR though.

Reply:
Added,